controllers/result.go

```diff
@@ -27,13 +32,23 @@ func (c *ResultController) List() {
 	c.TplName = "results/list.html"
 	web.ReadFromRequest(&c.Controller)
 
-	results, err := models.GetResultsByUserId(c.CurrentUser.Id)
+	totalResultCount, err := models.CountResultsByUserId(c.CurrentUser.Id)
 	if err != nil {
-		logs.Warn("Failed to get current user results: ", err.Error())
+		logs.Warn("Failed to count user results: ", err.Error())
 		c.Data["results"] = []*models.Result{}
 	}
 
-	c.Data["results"] = results
+	paginator := pagination.SetPaginator((*context.Context)(c.Ctx), ITEMS_PER_PAGE, totalResultCount)
+
+	results, err := models.GetPaginatedResultsByUserId(c.CurrentUser.Id, ITEMS_PER_PAGE, int64(paginator.Offset()))
+	if err != nil {
+		logs.Warn("Failed to get current user results: ", err.Error())
+	}
+
+	resultSets := helpers.PrepareResultSet(results)
+
+	c.Data["paginator"] = paginator
```

Comment:
Do we need to add it? I think Biggo automatically makes `paginator ` available in the view